lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -80,9 +83,20 @@ def pass_hook_checks(self, event: EventBase) -> bool:
 
         return True
 
+    def set_blocked_status_impossible_integration(self) -> None:
```

Comment:
The core condition of this method is duplicated between 3 methods: `set_blocked_status_impossible_integration`, `pass_hook_checks` and `get_invalid_integration_status`.
 
Could we change this method (set_blocked...) to something along the lines of: `is_valid_mongos_integration -> bool` and take out the status setting to `_on_relation_changed` directly, and we refer to it in the 2 other methods relying on it

Reply:
good idea, ty